### PR TITLE
Aave v2 connector

### PR DESCRIPTION
**Pull request by mubaris**
@thrilok209

### CODE REVIEW

**Comment by KaymasJain** [comment]
@mubaris @thrilok209 I was thinking to remove the common event thing for all our further connectors from now on to save gas.

**Comment by thrilok209** [comment]
@mubaris I did some code refactoring, could you check if everything looks?

File: contracts/connectors/aave_v2.sol [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+        AaveInterface aave = AaveInterface(getAaveProvider().getLendingPool());
+
+        bool isEth = token == getEthAddr();
+        address _token = isEth ? getWethAddr() : token;
+
+        TokenInterface tokenContract = TokenInterface(_token);
+
+        if (isEth) {
+            _amt = _amt == uint(-1) ? address(this).balance : _amt;
```
Comment (thrilok209) [inline contracts/connectors/aave_v2.sol]: @mubaris I think over here, `unit(-1)` case should be maxed payback. So here, we need to take the borrowed debt balance as the max value. 

Like this: 
https://github.com/InstaDApp/dsa-connectors/blob/09973b9221be2266dc72e047675ffdfbc0482f56/contracts/connectors/aave.sol#L177

File: contracts/connectors/aave_v2.sol [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+    function withdraw(address token, uint amt, uint getId, uint setId) external {
+        uint _amt = getUint(getId, amt);
+
+        AaveInterface aave = AaveInterface(getAaveProvider().getLendingPool());
+        bool isEth = token == getEthAddr();
+        address _token = isEth ? getWethAddr() : token;
+
+        TokenInterface tokenContract = TokenInterface(_token);
+
```
Comment (thrilok209) [inline contracts/connectors/aave_v2.sol:160]: @mubaris We need to add `uint(-1)` case over. Which would withdraw the maximum supplied amount. 

Like this: 
https://github.com/InstaDApp/dsa-connectors/blob/09973b9221be2266dc72e047675ffdfbc0482f56/contracts/connectors/aave.sol#L132

`BalanceOf` in the above snippet means, the balance of atoken(Which includes supplied interest)

File: contracts/connectors/aave_v2.sol [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        uint initialBal = tokenContract.balanceOf(address(this));
+        aave.withdraw(_token, _amt, address(this));
```
Comment (thrilok209) [inline contracts/connectors/aave_v2.sol:167]: @mubaris When `_amt` is `uint(-1)`, does it withdraw max amount supplied?

Comment (mubaris) [reply contracts/connectors/aave_v2.sol:167]: yes
image

File: contracts/connectors/aave_v2.sol (hunk 1 above)
Comment (mubaris) [reply contracts/connectors/aave_v2.sol]: @thrilok209 check now

**Comment by thrilok209** [comment]
LGTM, We can merge it once Aave V2 is live on mainnet.

**Review by thrilok209 — APPROVED** [review]
LGTM

**Pull request merged**